server/controllers/authController.js

```diff
@@ -0,0 +1,58 @@
+import passport from 'passport';
+import jwt from 'jsonwebtoken';
+
+const { JWT_EXPIRES_MS, JWT_SECRET } = process.env;
+/**
+ * This class contains all methods
+ * required to handle all
+ * authentication routes.
+ */
+class AuthController {
+  /**
+     * This method handle the sign request.
+     * @param {object} req The user's request.
+     * @param {object} res The response.
+     * @returns {object} The status and some data of the user.
+     */
+  static signIn(req, res) {
+    passport.authenticate(
+      'local',
+      { session: false },
+      (error, user) => {
+        if (error || !user) {
+          res.status(404).json({ error });
+        }
+
+        const { id, email, lastname } = user;
+        /** This is what goes in our JWT */
+        const payload = {
+          id,
+          expires: Date.now() + parseInt(JWT_EXPIRES_MS, 4),
+        };
+
+        /** assigns payload to req.user */
+        req.login(payload, { session: false }, (err) => {
+          if (err) {
+            res.status(400).send({ err });
+          }
```

Comment:
i think this good but it can be better if you add them in middle ware so that everyone can have access on the signed in user using that middle ware.

Reply:
@YankyJayChris  thanks for your feedback,
If I got it right, You want me to separate a common functionality like
Generating token, I can do that, but this is only signing in functionality, I don't think you will need to access inside functions. if you do please feel free to use this static method `signIn`
otherwise, if you see any other function that you need to use, please suggest it!